seaworthy/tests-pytest/test_pytest.py

```diff
@@ -0,0 +1,218 @@
+import docker
+import pytest
+
+from seaworthy.checks import docker_client
+from seaworthy.containers.base import ContainerBase
+from seaworthy.dockerhelper import DockerHelper, fetch_images
+from seaworthy.pytest.checks import dockertest
+from seaworthy.pytest.fixtures import (
+    clean_container_fixtures, container_fixture, docker_helper_fixture,
+    image_pull_fixture)
+
+
+IMG = 'nginx:alpine'
+
+
+def setup_module():
+    with docker_client() as client:
+        fetch_images(client, [IMG])
+
+
+# We redefine the docker_helper fixture here to have function scope so our
+# tests don't conflict.
+docker_helper = docker_helper_fixture(scope='function')
+
+
+@dockertest()
+class TestDockerHelperFixture:
+    def test_fixture(self, testdir):
+        """
+        The ``docker_helper`` fixture should automatically be in scope for a
+        project that uses this pytest plugin.  When the ``DockerHelper`` is
+        passed to a test function using a fixture, it should be possible to
+        create, start, stop, and remove a container using the helper.
+        """
+        testdir.makepyfile("""
+            from docker.errors import NotFound
+            import pytest
+
+            def test_container_basics(docker_helper):
+                container = docker_helper.create_container('test', '{}')
+                docker_helper.start_container(container)
+
+                assert container.status == 'running'
+
+                docker_helper.stop_and_remove_container(container)
+
+                with pytest.raises(NotFound):
+                    container.reload()
+        """.format(IMG))
+
+        result = testdir.runpytest()
+        result.assert_outcomes(passed=1)
+
+    def test_setup_teardown(self):
+        """
+        The fixture should yield a setup helper, and afterwards tear down the
+        helper.
+        """
+        fixture_gen = docker_helper()
+        helper = next(fixture_gen)
+        assert isinstance(helper, DockerHelper)
+
+        # Test we can create a container; if we can the helper must be set
+        # up
+        container = helper.create_container('test', IMG)
```

Comment:
The Docker client container methods can take `Image` objects or tag strings (although this is not documented). Our `DockerHelper` can only take tag strings, though.

Reply:
Do we expect people to pass the return value of `fetch_image()` to these?

Reply:
Maybe, now that it's easier to fetch Image objects.

An `Image` object suggests (although doesn't guarantee) that the image is present on the system. It might also be nice to use something a bit more formal than a string.